loki/transformations/build_system/dependency.py

```diff
@@ -329,6 +333,34 @@ def rename_imports(self, source, imports, targets=None):
         if import_map:
             source.spec = Transformer(import_map).visit(source.spec)
 
+    def rename_access_specs(self, module, targets=None, active_nodes=None):
+        """
+        Update/rename access specifiers.
+
+        Parameters
+        ----------
+        module : :any:`Module`
+            The IR object to transform
+        targets : list of str
+            Optional list of subroutine names for which to modify access specs
+        active_nodes : list of str
+            Optional list of active nodes
+        """
+        if module.public_access_spec:
+            if active_nodes is not None:
+                new_access_spec = tuple(elem for elem in module.public_access_spec if elem in active_nodes)
+            else:
+                new_access_spec = module.public_access_spec
+            module.public_access_spec = tuple(f'{elem}{self.suffix}' if not targets or elem in targets else
+                    elem for elem in new_access_spec)
+        if module.private_access_spec:
+            if active_nodes is not None:
+                new_access_spec = tuple(elem for elem in module.private_access_spec if elem in active_nodes)
+            else:
+                new_access_spec = module.public_access_spec
+            module.private_access_spec = tuple(f'{elem}{self.suffix}' if not targets or elem in targets
+                    else elem for elem in new_access_spec)
```

Comment:
The docstring doesn't tell much... But I would make this utility even more basic, something like
```suggestion
    def rename_access_spec_names(self, access_spec, targets=None, active_nodes=None):
        """
        Rename target names in an access spec
        
        For all names in the access spec that are contained in :data:`targets`, rename them as 
        ``{name}{self.suffix}``. If :data:`active_nodes` are given, then all names
        that are not in the list of active nodes, are being removed from the list.

        Parameters
        ----------
        access_spec : list of str
            List of names from an access spec
        targets : list of str
            Optional list of subroutine names for which to modify access specs
        active_nodes : list of str
            Optional list of active nodes
        """
        if active_nodes:
            access_spec = tuple(elem for elem in access_spec if elem in active_nodes)
        return tuple(
            f'{elem}{self.suffix}' if not targets or elem in targets
            else elem
            for elem in access_spec
        )
```

And then use that utility simply as:
```python
if module.public_access_spec:
    module.public_access_spec = self.rename_access_spec(
        module.public_access_spec, targets=targets, active_nodes=active_nodes
    )

if module.private_access_spec:
    module.private_access_spec = self.rename_access_spec(
        module.private_access_spec, targets=targets, active_nodes=active_nodes
    )
```